cors/access-control-expose-headers-parsing.window.js

```diff
@@ -1,13 +1,15 @@
-function exposeTest(resource, desc) {
-  const url = new URL("resources/" + resource, location.href).href.replace("://", "://élève.");
+promise_test(() => fetch("resources/access-control-expose-headers.json").then(res => res.json()).then(runTests), "Loading JSON…");
 
-  promise_test(() => {
-    return fetch(url).then(res => {
-      assert_equals(res.headers.get("content-language"), "sure")
-      assert_equals(res.headers.get("x-custom"), null);
-    })
-  }, "Access-Control-Expose-Headers parsing: " + desc);
+function runTests(allTestData) {
+  allTestData.forEach(testData => {
+    const encodedInput = encodeURIComponent(testData.input);
+    promise_test(() => {
```

Comment:
So these tests would be defined quite late (not after first-run). We wanted to not really do those if we could earlier. Often we'd define the tests, and only later do the running things.

But I can understand that things might have changed.

Why are you defining the tests in JSON:
```
[
  {
    "input": "access-control-expose-headers: BB-8",
    "exposed": true
  },
  {
    "input": "Access-Control-Expose-Headers: bb-8,,@#$#%%&^&^*()()11!",
    "exposed": false
  },
```
Instead of JS (where the tests would not have been runtime-defined):
```
runTest({input: "access-control-expose-headers: BB-8", exposed: true});
runTest({input: "Access-Control-Expose-Headers: bb-8,,@#$#%%&^&^*()()11!", exposed: false});
```
Or maybe there's some other system reading and defining test from the JSON? I see there's some very few tests defined like this already, but the other pattern is much more used still.

---
Could even inline the JSON in the test to get around "tests only defined after fetch has finished".

Reply:
Okay :)